applications/asset_tracker_v2/src/cloud/nrf_cloud_integration.c

```diff
@@ -253,8 +252,7 @@ static int decode_and_send(cJSON *object, const char *object_name, bool device_s
 
 		if (tx_buffer == NULL) {
 			LOG_ERR("Failed to allocate memory for JSON string");
-			err = -ENOMEM;
-			goto exit;
+			return -ENOMEM;
```

Comment:
cJSON_Delete(msg_ref) no longer called in this case. 
Is this intentional?

Reply:
Yes. Deletion of this object was moved to line 237 to avoid a potential double free.

Reply:
In the case of `device_state` being `true` we associate `msg_ref` to `state_obj`. In this case we only have to free `state_obj` due to the object being associated with a parent. If `device_state` evaluates to `false` the object is deleted.

Reply:
I think we need to propagate the return value from `cJSON_AddItemToObject()` that's wrapped by `json_add_obj()`. If the return is `false`, `msg_ref` was not added to `state_obj`, and then we have a leak

Reply:
Good point. There are quite a few instances of both `json_add_obj()` and `json_add_obj_array` throughout the app. Ill add a follow up PR that addresses this issue for all cases.